src/fireedge/src/server/routes/api/oneprovision/provider/functions.js

```diff
@@ -0,0 +1,430 @@
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+
+const { parse } = require('yaml')
+const { basename } = require('path')
+const { defaults, httpCodes } = require('server/utils/constants')
+const {
+  httpResponse,
+  parsePostData,
+  getFilesbyEXT,
+  existsFile,
+  executeCommand,
+  removeFile,
+} = require('server/utils/server')
+const {
+  createTemporalFile,
+  createYMLContent,
+  getEndpoint,
+  getSpecificConfig,
+} = require('server/routes/api/oneprovision/utils')
+
+const {
+  defaultFolderTmpProvision,
+  defaultCommandProvider,
+  defaultHideCredentials,
+  defaultHideCredentialReplacer,
+  defaultEmptyFunction,
+  defaultProvidersConfigPath,
+} = defaults
+const { ok, internalServerError } = httpCodes
+const httpInternalError = httpResponse(internalServerError, '', '')
+
+/**
+ * Get provision Config.
+ *
+ * @param {object} res - http response
+ * @param {Function} next - express stepper
+ * @param {object} params - params of http request
+ * @param {object} userData - user of http request
+ */
+const getProviderConfig = (
+  res = {},
+  next = defaultEmptyFunction,
+  params = {},
+  userData = {}
+) => {
+  const path = `${global.paths.PROVISION_PATH}${defaultProvidersConfigPath}`
+  const extFiles = 'yaml'
+  let response = internalServerError
+  let message = ''
+
+  /**
+   * Fill Providers.
+   *
+   * @param {string} content - content of provider
+   * @param {string} name - name of provider
+   */
+  const fillProviders = (content = '', name = '') => {
+    if (content && name) {
+      try {
+        message[name] = parse(content)
+      } catch (error) {}
+    }
+  }
+  /**
+   * Fill Error Message.
+   *
+   * @param {string} error - error path
+   */
+  const fillErrorMessage = (error = '') => {
+    message = error
+  }
+
+  const files = getFilesbyEXT(path, extFiles, fillErrorMessage)
+
+  if (files.length) {
+    response = ok
+    files.forEach((file) => {
+      existsFile(file, (content) => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).